snapcraft/internal/pluginhandler/__init__.py

```diff
@@ -1130,7 +1130,13 @@ def _organize_filesets(part_name, fileset, base_dir, overwrite):
 
 def _clean_migrated_files(snap_files, snap_dirs, directory):
     for snap_file in snap_files:
-        os.remove(os.path.join(directory, snap_file))
+        try:
+            os.remove(os.path.join(directory, snap_file))
+        except FileNotFoundError:
+            logger.warning(
+                "File {name!r} not found. Did you remove it in a different "
+                "part?".format(name=snap_file)
```

Comment:
Asking this question is rubbing me the wrong way. I'd prefer the warning just stated the problem instead of guessing why the problem exists. There are other ways this happens, too (e.g. manually removing files or moving them around). I suggest rephrasing these warnings to be statements, for example: "Attempted to remove file {!r}, but it didn't exist. Skipping..."

Reply:
I had this in my draft notes, then saw you reviewed and did not bring it up so I thought I was being picky :smile:. I would like to reuse the error pattern for warnings as much as possible https://github.com/snapcore/snapcraft/blob/master/CODE_STYLE.md#errors

Reply:
Agreed. That's something I considered for the same reasons, but (reluctantly) ended up with the question because it was stylistically similar to another message already in the code. Given the uncertainty/guessing factor there, the fact statement is indeed a better option. New patch on the way.

Reply:
> That's something I considered for the same reasons, but (reluctantly) ended up with the question because it was stylistically similar to another message already in the code.

Trust your gut. Parts of this codebase have been around for a _long_ time, and pluginhandler is one of the worst offenders. Established patterns aren't always the best, if you feel something isn't quite the way it should be, always feel free to say so!